Listenbrainz/ViewModel/SearchViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SearchViewModel.swift
+//  Listenbrainz
+//
+//  Created by Gaurav Bhardwaj on 03/06/24.
+//
+
+import Combine
+import SwiftUI
+
+class SearchViewModel: ObservableObject {
+    @Published var searchTerm: String = ""
+    @Published var users: [User] = []
+    @Published var isActive: Bool = false
+    @Published var error: String? = nil
+
+    private var cancellable: AnyCancellable?
+
+    init() {
+        cancellable = $searchTerm
+            .debounce(for: .milliseconds(500), scheduler: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] term in
+                self?.searchUsers(term: term)
+            }
+    }
+
+    func searchUsers(term: String) {
+        guard !term.isEmpty else {
+            self.users = []
+            return
+        }
+
+        let urlString = "\(BuildConfiguration.shared.API_LISTENBRAINZ_BASE_URL)/search/users?search_term=\(term)"
+        guard let url = URL(string: urlString) else { return }
+
+        URLSession.shared.dataTask(with: url) { data, response, error in
```

Comment:
This is incorrect, use Alamofire and the repositories here.

Reply:
done